src/matches.ts

```diff
@@ -56,7 +56,7 @@ function matches(
   if (matcher instanceof Function) {
     return matcher(normalizedText, node)
   } else if (matcher instanceof RegExp) {
-    return matcher.test(normalizedText)
+    return new RegExp(matcher).test(normalizedText)
```

Comment:
does this preserve all flags? I didn't look at our existing tests so if we already have tests for all flags then this is good to go.

Reply:
Good catch, the tests on `matches` don't have any flags, but indirectly a few flags are used throughout the test suite. I'll provide more test cases for `matches`.

Reply:
Honestly I would just throw on regexp with a global flag (if we can detect that). It doesn't make sense to pass anyway and complicates things for us needlessly

Reply:
We can detect if it has a global flag.

Throwing would result in a new major version. Even a fix like the proposed one, or another one, would also lead to a new major because the behaviour will change. 

The more I think abou it, doing nothing and keeping the current implementation might be the best here. We could add a note to the docs. Using a global flag is rare and it doesn't do anything for these cases.

Reply:
> Throwing would result in a new major version.

I don't see it that way. The old behavior with global flags was clearly wrong anyway. We can do `console.warn` though to make it clearer. 

> Using a global flag is rare and it doesn't do anything for these cases.

Note that a global flag does something as is apparent by the issue. It returns the wrong number of results which is a bug.

Reply:
I agree that it's wrong, but I can imagine at there are a few global regexes to just find a single element. This currently works, but would result that it throws in a new version. 

After looking at the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#instance_properties), `lastIndex` is only used for globals, so that would be a safe fix.

I would prefer that over throwing, `console.warn` is also fine for me - this could also become stricter (as in an error) in the future when there are more changes to create a new major.


> Note that a global flag does something as is apparent by the issue. It returns the wrong number of results which is a bug.

True, I formulated the sentence badly. What I meant was that to find an element it doesn't matter if a global flag is added to the regex. 


I've updated the commit with the `lastIndex` implementation, if you prefer the `console.warn` let me know and I'll change it. Thanks for your help.